src/journey_to_response.cc

```diff
@@ -341,7 +341,14 @@ api::Itinerary journey_to_response(
                       exit_stop.scheduled_time(n::event_type::kArr),
                   .realTime_ = fr.is_rt(),
                   .scheduled_ = fr.is_scheduled(),
-                  .headsign_ = std::string{enter_stop.direction()},
+                  .headsign_ =
+                      [&]() {
+                        if (!enter_stop.direction().empty()) {
+                          return std::string{enter_stop.direction()};
+                        }
+
+                        return to_place(fr[t.stop_range_.to_ - 1], false).name_;
+                      }(),
```

Comment:
```suggestion
                  .headsign_ = enter_stop.direction().empty()
                                 ? to_place(fr[t.stop_range_.to_ - 1], false).name_
                                 : std::string{enter_stop.direction()}
```

No need for a complicated lambda if `?:` does the job. I didn't format it though.